torch/csrc/autograd/function.h

```diff
@@ -115,6 +115,12 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
     RECORD_FUNCTION(
         this, std::vector<c10::IValue>(inputs.begin(), inputs.end()));
 
+#ifdef BUILD_NAMEDTENSOR
+    // In the first iteration of named tensors, autograd ignores names and
+    // operates on unnamed tensors. In the long term, autograd should
+    // probably operate with names.
+    at::NoNamesGuard no_names_guard;
```

Comment:
@ezyang is it guaranteed that all calls to autograd function go through `operator()`? I tried looking through the code and I think this is what happens.

Reply:
I think so. If you want to be 100% sure you could make the apply() function virtual private, which will catch at compile time if anyone else naughtily calling apply().